### PR TITLE
Laboratory work #1, Doroshina Aleksandra - 23FPL1

### PR DESCRIPTION
**Pull request by DoroshinaAleksandra**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -34,7 +45,20 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
-
+    if isinstance(tokens, list):
+        for element in tokens:
+            if isinstance(element, str):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:50]: THe same

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -34,7 +45,20 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+                number_of_tokens = len(tokens)
+                tokens_quantity = {}
+                for letter in tokens:
+                    if letter not in tokens_quantity:
+                        tokens_quantity[letter] = 1
+                    else:
+                        tokens_quantity[letter] = tokens_quantity[letter] + 1
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:57]: use `+=1`

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -34,7 +45,20 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
     In case of corrupt input arguments, None is returned
     """
-
+    if isinstance(tokens, list):
+        for element in tokens:
+            if isinstance(element, str):
+                number_of_tokens = len(tokens)
+                tokens_quantity = {}
+                for letter in tokens:
+                    if letter not in tokens_quantity:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:54]: if not - set 0 and do `+= 1` always

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -49,7 +73,10 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
 
     In case of corrupt input arguments, None is returned
     """
-
+    if isinstance(language, str):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:76]: the same strategy

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -67,8 +94,8 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
 
 
 def compare_profiles(
-    unknown_profile: dict[str, str | dict[str, float]],
-    profile_to_compare: dict[str, str | dict[str, float]],
+        unknown_profile: dict[str, str | dict[str, float]],
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:97]: revert this chanege

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -87,9 +114,9 @@ def compare_profiles(
 
 
 def detect_language(
-    unknown_profile: dict[str, str | dict[str, float]],
-    profile_1: dict[str, str | dict[str, float]],
-    profile_2: dict[str, str | dict[str, float]],
+        unknown_profile: dict[str, str | dict[str, float]],
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:117]: revert this change too

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -152,7 +179,7 @@ def collect_profiles(paths_to_profiles: list) -> list[dict[str, str | dict[str,
 
 
 def detect_language_advanced(
-    unknown_profile: dict[str, str | dict[str, float]], known_profiles: list
+        unknown_profile: dict[str, str | dict[str, float]], known_profiles: list
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:182]: revert this change too

File: lab_1_classify_by_unigrams/start.py [hunk 8]
```diff
@@ -8,13 +8,15 @@ def main() -> None:
     """
     Launches an implementation
     """
+    from main import create_language_profile
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/start.py:11]: move this import to the top of the file

File: seminars/practice_1_run_debug.py [hunk 9]
```diff
@@ -1,3 +1,13 @@
+'''a = 0
```
Comment (artyomtugaryov) [inline seminars/practice_1_run_debug.py:1]: revert seminars

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Please, fix my comments + change the target score

File: lab_1_classify_by_unigrams/assets/texts/en.txt [hunk 10]
```diff
@@ -14,4 +14,4 @@ direction as the sun. However, if, for example, the feeding place is 40º to the
 the sun, then the dancer would run 40º to the left of the vertical line. This was to be
 the first of von Frisch’s remarkable discoveries. Soon he would also discover a
 number of other remarkable facts about how bees communicate and, in doing so,
-revolutionise the study of animal behaviour generally.
+revolutionise the study of animal behaviour generally.
```
[the remark this answers is not in the document]
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/assets/texts/en.txt]: done

File: lab_1_classify_by_unigrams/main.py [hunk 11]
```diff
@@ -3,6 +3,8 @@
 
 Language detection
 """
+
```
[the remark this answers is not in the document]
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py [hunk 12]
```diff
@@ -20,6 +22,15 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    tokenized_text = []
+    if isinstance(text, str):
```
[the remark this answers is not in the document]
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py [hunk 13]
```diff
@@ -34,7 +45,20 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
-
+    if isinstance(tokens, list):
```
[the remark this answers is not in the document]
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 4 above)
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 6 above)
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/start.py (hunk 8 above)
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/start.py]: done

File: seminars/practice_1_run_debug.py (hunk 9 above)
Comment (DoroshinaAleksandra) [reply seminars/practice_1_run_debug.py]: done

File: lab_1_classify_by_unigrams/main.py [hunk 14]
```diff
@@ -64,6 +93,21 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
 
     In case of corrupt input arguments, None is returned
     """
+    """
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:96]: does it work for you?

File: lab_1_classify_by_unigrams/main.py [hunk 15]
```diff
@@ -84,6 +128,39 @@ def compare_profiles(
     In case of corrupt input arguments or lack of keys 'name' and
     'freq' in arguments, None is returned
     """
+    """
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:131]: just remove if  you do not want to continue for 8

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 14 above)
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py:96]: Unfortunately, it does not. I tried to fix it, but unsuccessfully

File: lab_1_classify_by_unigrams/main.py (hunk 15 above)
Comment (DoroshinaAleksandra) [reply lab_1_classify_by_unigrams/main.py:131]: OK
